dev/core#5580 Fix ActionSchedule save on repeat events

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
_A brief description of the pull request. Keep technical jargon to a minimum. Hyperlink relevant discussions._

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

eileenmcnaughton committed:
Fix ActionSchedule save on repeat events

seamuslee001 commented:
@eileenmcnaughton I tried this out and when creating a brand new recuring series i got this error

![image](https://github.com/user-attachments/assets/7b201d92-45ab-4e9d-a359-e9658e0c20fb)

eileenmcnaughton commented:
Hmm - well at least some of those should not be mandatory...

eileenmcnaughton committed:
Title is only required if name is false

eileenmcnaughton committed:
Make mapping_id & start_action_unit conditionally required

The fact they are not set for repeating events means they are not truly required

eileenmcnaughton commented:
Ok - title is not required when name is present - fixed

The other 2 are clearly non mandatory for events as I confirmed they were being left empty prior - fixed also

eileenmcnaughton committed:
Also conditionally permit start action condition

eileenmcnaughton committed:
Fix deprecation when NULL passed to explode

seamuslee001 commented:
OK this seems to work in my local testing
